### PR TITLE
Certificate of Eligibility | Update to web components

### PR DESCRIPTION
**Pull request by Mottie**
## Original Ticket

[#47111](https://github.com/department-of-veterans-affairs/va.gov-team/issues/47111)

## URL of change

- `/housing-assistance/home-loans/request-coe-form-26-1880/introduction`
- `/housing-assistance/home-loans/check-coe-status/your-coe/`

## Background

COE was still using React Components that need to be updated to use the equivalent Web Component.

While modifying code, a problem with the UX was discovered (see https://github.com/department-of-veterans-affairs/va.gov-team/issues/47469) which needs to be addressed, but this PR did include some changes to include already submitted files within the success message

After changing `FileInput` to `va-file-input`, updating the unit tests proved to be difficult due to testing-library/react not supporting testing the shadow DOM, making it difficult (impossible?) to click/trigger an upload. So use of the `FileInput` React component was restored

## Steps to test

1. Pull in this branch locally, or test in the review instance
2. Log in with user 228 (Mark Webb)
3. Go to the first URL of change
4. Check that the web components are in place
5. Go to the second URL of change
6. Check that the web components are in place - you won't be able to test the uploader without manipulating the code

### Form app
- `OMBInfo` → `va-omb-info`
- `va-telephone` → add `tty` prop
- Download link → `va-link`

### Status app
- `TextInput` → `va-text-input`
- `Select` → `va-select`
- Download link → `va-link` & remove static file size value. Pull file extension from file name (`mimeType`).
- Fix date within download links from `MMDDYYY` to `MM-DD-YYYY` to make it more readable
- Link to start COE form → action link
  <details><summary>COE status page</summary>
  COE status with proper download links with icons and a blue action link to start the form
  </details>
- Include submitted files within the success message
  <details><summary>Upload success</summary>
  upload success message with 2 files that were just uploaded listed within
  </details>

## How was your code tested

Updated unit & e2e tests

## Acceptance criteria
- [x] Update to use web components
- [x] Fix download link formatting
- [x] Minor a11y & UX fixes
- [x] Add unit tests
- [x] All tests passing

## Definition of done
- [ ] Events are logged appropriately
- [ ] Documentation has been updated, if applicable
- [x] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

File: src/applications/lgy/coe/status/components/DocumentUploader/FileListItem.jsx [hunk 1]
```diff
@@ -11,7 +11,7 @@ const FileListItem = ({ index, file, onClick }) => {
 
   return (
     <div
-      className="vads-u-background-color--gray-lightest vads-u-padding-y--1 vads-u-padding-x--2 vads-u-margin-y--1"
+      className="file-list-item vads-u-background-color--gray-lightest vads-u-padding-y--1 vads-u-padding-x--2 vads-u-margin-y--1"
```
Comment (jerekshoe) [inline src/applications/lgy/coe/status/components/DocumentUploader/FileListItem.jsx:14]: Is this supposed to have been added? I'm not seeing it used as a selector or anything

Comment (Mottie) [reply src/applications/lgy/coe/status/components/DocumentUploader/FileListItem.jsx:14]: Yeah, I was using it in a unit test, but ended up changing it. I'll need to remove this class then.

File: src/applications/lgy/coe/form/content/LoggedInContent.jsx [hunk 2]
```diff
@@ -1,5 +1,4 @@
 import React from 'react';
-import OMBInfo from '@department-of-veterans-affairs/component-library/OMBInfo';
```
Comment (rocio-desantiago) [inline src/applications/lgy/coe/form/content/LoggedInContent.jsx:2]: Is OMBInfo still being used in other files or can we go ahead and remove it from our component-library?

File: src/applications/lgy/coe/form/content/LoggedInContent.jsx [hunk 3]
```diff
@@ -37,7 +36,7 @@ const LoggedInContent = ({ route }) => (
         headingLevel={2}
       />
     </div>
-    <OMBInfo expDate="11/30/2022" ombNumber="2900-0086" resBurden={15} />
+    <va-omb-info exp-date="11/30/2022" omb-number="2900-0086" res-burden={15} />
```
Comment (rocio-desantiago) [inline src/applications/lgy/coe/form/content/LoggedInContent.jsx:39]: does the component `<va-omb-info/>` this automatically get imported?

File: src/applications/lgy/coe/shared/components/ReviewAndDownload.jsx [hunk 4]
```diff
@@ -20,7 +16,13 @@ const ReviewAndDownload = () => (
       </a>
     </p>
     <div className="vads-u-margin-top--4">
-      <DownloadLink href={downloadUrl} label={downloadLinkLabel} />
+      <va-link
+        download
+        filetype="PDF"
+        pages="1"
+        href={downloadUrl}
+        text="Download your COE"
```
Comment (rocio-desantiago) [inline src/applications/lgy/coe/shared/components/ReviewAndDownload.jsx:24]: Was there a decision to remove `(PDF) 0.20MB` from the textlabel?

File: src/applications/lgy/coe/status/components/DocumentList/List.jsx [hunk 5]
```diff
@@ -8,30 +8,33 @@ import { formatDateLong } from 'platform/utilities/date';
...
-    timestamp,
-  )} (${getDocumentType(documentType)})`;
+export const getDownloadLinkLabel = timestamp =>
+  `Download Notification Letter ${formatLabelDate(timestamp)}`;
 
 const List = ({ documents }) =>
   documents.map((document, i) => {
-    const { createDate, description, documentType, id } = document;
-    const downloadLinkLabel = getDownloadLinkLabel(createDate, documentType);
+    const { createDate, description, mimeType, id } = document;
```
Comment (rocio-desantiago) [inline src/applications/lgy/coe/status/components/DocumentList/List.jsx:28]: what's a mimeType?

**Review by rocio-desantiago — COMMENTED** [review]
Just some questions.

**Review by rocio-desantiago — APPROVED** [review]
ran locally and components are showing up as expected

File: src/applications/lgy/coe/form/content/LoggedInContent.jsx (hunk 2 above)
Comment (Mottie) [reply src/applications/lgy/coe/form/content/LoggedInContent.jsx:2]: This is the only file that uses `OMBInfo`

File: src/applications/lgy/coe/form/content/LoggedInContent.jsx (hunk 3 above)
Comment (Mottie) [reply src/applications/lgy/coe/form/content/LoggedInContent.jsx:39]: All web components (components starting with `va-`) get imported by the main component (the one that sets up Redux)

File: src/applications/lgy/coe/status/components/DocumentList/List.jsx (hunk 5 above)
Comment (Mottie) [reply src/applications/lgy/coe/status/components/DocumentList/List.jsx:28]: LGY doesn't have this named properly. A [proper mime type](https://developer.mozilla.org/en-US/docs/Web/HTTP/Basics_of_HTTP/MIME_types/Common_types) for PDFs would be `application/PDF`, but in this case the `mimeType` contains the full file name with extension.

**Pull request merged**